main/src/library/flix/actor/ActorSystem.flix

```diff
@@ -143,25 +155,36 @@ namespace ActorSystem {
   }
 
   ///
-  /// Runs the shutdown policy of an actor system. Directly returns the updated actor map after one iteration of the policy.
+  /// Runs the shutdown policy of an actor system.
+  /// Returns the updated actor map after one iteration of the policy, or None if the system should stop.
   ///
-  def runShutdownPolicy(s: ActorSystem, actors: ActorMap, _control: Channel[SystemEvent]): ActorMap = {
+  def runShutdownPolicy(s: ActorSystem, actors: ActorMap, _control: Channel[SystemEvent], notifyChan: Channel[Unit]): Option[ActorMap] = {
```

Comment:
As a high-level comment, can we articulate why the actor system itself should have three states?

After all, can an actor not crash during startup/shutdown? And if so, what is our view on that.

Reply:
For shutdown, at least, it does make sense to talk about a system where it is true that every actor has terminated (c.f. structured concurrency/nurseries) so there I can somewhat buy the need for some special behavior.

For an actor system in the process of starting and one that is already started it seems less clear to me.

Reply:
While it is necessary to have a specific shutdown state (because when the policy is "complete", the actor system stops), the start state is just a subset of the running state. 
It would simplify things to have only a running and a shutdown state. This would however imply that start and restart policies will have to be composed together (which does not seem to be a problem).

Reply:
Do you think we should do that? I am on the fence.

Reply:
I think it is doable within the implementation without affecting the definition of policies themselves (they would be kept separate). So it would be transparent to the user, and should not change the overall semantics of an actor system. The main impact then is a simplification of the implementation.

Reply:
I quickly tried to merge start and restart policies, but the system does not start correctly anymore. So it might not be that simple. We can probably have a more in-depth look at that later.